back-end/hub-core/src/main/java/io/apicurio/hub/core/registry/http/HttpRegistry.java

```diff
@@ -24,74 +18,49 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 @ApplicationScoped
 @Default
 public class HttpRegistry implements IRegistry {
 
-    private static final Logger logger = LoggerFactory.getLogger(HttpRegistry.class);
-    private static CloseableHttpClient httpClient = HttpClients.createDefault();
-    private static final ObjectMapper mapper = new ObjectMapper();
+    private final RegistryRestClient registryRestClient;
 
     @Inject
-    private HubConfiguration config;
-    
+    public HttpRegistry(HubConfiguration config) {
+        this.registryRestClient = Optional.ofNullable(config.getRegistryUrl())
+                .map(url -> url.endsWith("/") ? url + "api" : url + "/api")
```

Comment:
I think this should be removed, and anyone configuring the system should provide the full endpoint to the root of the API.  The reason is that the next version of Apicurio Registry will be moving around the endpoints a bit, due to a forthcoming "v2" REST API.  So there will be two endpoints:  `/api/v1` and `/api/v2` - with that change it will become impossible to properly configure the endpoint here.